src/core/thread/mle_types.hpp

```diff
@@ -567,6 +568,61 @@ class LeaderData : public otLeaderData
     void SetLeaderRouterId(uint8_t aRouterId) { mLeaderRouterId = aRouterId; }
 };
 
+OT_TOOL_PACKED_BEGIN
+class RouterIdSet
+{
+public:
+    /**
+     * This method clears the Router Id Set.
+     *
+     */
+    void Clear(void) { memset(mRouterIdSet, 0, OT_ARRAY_LENGTH(mRouterIdSet)); }
```

Comment:
```suggestion
    void Clear(void) { memset(mRouterIdSet, 0, sizeof(mRouterIdSet)); }
```
I think better to use `sizeof()` since we care more about the size rather than the number of elements.